llvm/lib/Target/EraVM/EraVMTieSelectOperands.cpp

```diff
@@ -96,6 +96,17 @@ bool EraVMTieSelectOperands::tryPlacingTie(MachineInstr &MI,
   if (EraVM::out0Iterator(MI)->getReg().isPhysical())
     return false;
 
+  // Since RegisterCoalescer works well when we have both early clobber and
+  // explicit tied register, but not with implicit tied register, remove the
+  // early clobber flag so RegisterCoalescer can remove unneeded copies.
+  // This is safe to do because the regalloc will allocate the same register
+  // for the output and input register operand, so if there is a register in
+  // code operand, it won't be overwritten with the output register and we can
+  // do proper expansion PostRA.
+  if ((MI.getOpcode() == EraVM::SELcrr || MI.getOpcode() == EraVM::SELrcr) &&
```

Comment:
Nit: we might have not to put it for `SELcrr`, `SELrcr` in the first place, haven't we?

Reply:
Do you want me to remove opcode checking? We can do that, but I wanted this code to be in sync with .td file where we don't have tie constraint.

Reply:
I mean instead of placing clobbers on all selects and then conditionally remove them, we can conditionally place them. Right?

Reply:
We are placing early clobber flag only in case of select where register is used in stack/code operand in `EraVMBytesToCells`, and this is because of the functionality. Here, we are removing it because of for the reason described in the comment. If we want to remove this, we would need to run `LiveVariables` pass before `EraVMBytesToCells`, and skip adding early clobber flag if we are going to tie register operand for `SELcrr` and `SELrcr`. This will just add unnecessary complexity to the `EraVMBytesToCells` code, and I would like to avoid that.

Reply:
Got it. Thanks